theano/tensor/elemwise.py

```diff
@@ -502,6 +503,7 @@ class Elemwise(OpenMPOp):
     | Elemwise(log)(rand(3, 4, 5))
 
     """
+    __props__ = ("scalar_op", "inplace_pattern", "name", "nfunc_spec", "openmp")
```

Comment:
inplace_pattern is a dict, so it won't work with the current code. For now, I think you should concentrate on the cgt optimization, so just remove this line ( and don't add a warning for op without props)